$OpenBSD: patch-lib_Target_Sparc_SparcISelLowering_cpp,v 1.6 2020/08/05 06:49:48 jca Exp $

- Make sure that we really don't emit quad-precision unless the
  "hard-quad-float" feature is available
- Use a 64-bit register when required

Index: lib/Target/Sparc/SparcISelLowering.cpp
--- lib/Target/Sparc/SparcISelLowering.cpp.orig
+++ lib/Target/Sparc/SparcISelLowering.cpp
@@ -3105,6 +3105,11 @@ SparcTargetLowering::EmitInstrWithCustomInserter(Machi
   case SP::SELECT_CC_DFP_ICC:
   case SP::SELECT_CC_QFP_ICC:
     return expandSelectCC(MI, BB, SP::BCOND);
+  case SP::SELECT_CC_Int_XCC:
+  case SP::SELECT_CC_FP_XCC:
+  case SP::SELECT_CC_DFP_XCC:
+  case SP::SELECT_CC_QFP_XCC:
+    return expandSelectCC(MI, BB, SP::BPXCC);
   case SP::SELECT_CC_Int_FCC:
   case SP::SELECT_CC_FP_FCC:
   case SP::SELECT_CC_DFP_FCC:
